packages/shared/src/components/PostOptionsMenu.tsx

```diff
@@ -99,9 +109,26 @@ export default function PostOptionsMenu({
     });
     onRemovePost?.(_postIndex);
   };
-  const { onConfirmDeletePost } = usePostMenuActions({
+  const { onConfirmDeletePost, onPinPost } = usePostMenuActions({
     post,
     postIndex,
+    onPinSuccessful: async () => {
+      const key = getPostByIdKey(post.id);
+      const cached = client.getQueryData(key);
+      if (cached) {
+        client.setQueryData<Post>(key, (data) => ({
+          ...data,
+          pinnedAt: post.pinnedAt ? null : new Date(),
+        }));
+      }
+
+      await client.invalidateQueries(feedQueryKey);
+      displayToast(
+        post.pinnedAt
+          ? 'Your post has been unpinned'
+          : '📌 Your post has been pinned',
+      );
+    },
```

Comment:
Is this not something we can handle within the hook?
It seems like the wrong place to handle it per implementation.

Reply:
Have actually done that initially. But then I saw `PostOptionsMenu` handling all the toast and invalidations so I've thought of keeping everything in the same place. WDYT?